### PR TITLE
fix pod availability checking logic

### PR DESCRIPTION
**Pull request by Hsn723**
The changes to `IsPodAvailable` introduced in https://github.com/gruntwork-io/terratest/pull/480 incorrectly reported pods as available when the readinessProbe is failing but the containers were in a started state. Namely, when there is no startupProbe, `containerStatus.Started` will always be true ([ref](https://godoc.org/k8s.io/api/core/v1#ContainerStatus)) and the pod will be incorrectly reported as available.

This PR fixes the availability checking logic and adds a few test cases to validate the fix.

### CODE REVIEW

File: modules/k8s/pod.go [hunk 1]
```diff
@@ -142,7 +142,7 @@ func IsPodAvailable(pod *corev1.Pod) bool {
 		isContainerStarted := containerStatus.Started
 		isContainerReady := containerStatus.Ready
 
-		if !isContainerReady && isContainerStarted != nil && !*isContainerStarted {
+		if !isContainerReady || isContainerStarted == nil || !*isContainerStarted {
```
Comment (brikis98) [inline modules/k8s/pod.go:145]: NIT: `*isContainerStarted == false` is a tiny bit easier to read than `!*isContainerStarted`.

**Review by brikis98 — COMMENTED** [review]
Thanks for adding the test cases!

This fix seems correct to me, but @yorinasub17, I'd appreciate a second set of eyes.

**Review by infraredgirl — DISMISSED** [review]
LGTM!

**Review by yorinasub17 — DISMISSED** [review]
This LGTM as well!

**Comment by brikis98** [comment]
Thank you all for the second (and third) sets of eyes!

@Hsn723 Could you fix the one NIT I mentioned, and pull in the latest from `master`? I can kick off tests then.

**Comment by Hsn723** [comment]
@brikis98 Thank you for the reviews. I have applied the requested fixes and synced to `master`

**Review by brikis98 — DISMISSED** [review]
Great, thanks! I'll kick off tests now.

**Comment by brikis98** [comment]
Tests failed due to precommit hooks. Could you run `go fmt` on the code?

```
[INFO] Initializing environment for https://github.com/gruntwork-io/pre-commit.
goimports................................................................Failed
- hook id: goimports
- files were modified by this hook

modules/k8s/pod_test.go
```

**Comment by Hsn723** [comment]
@brikis98 You're right, I forgot to run gofmt. I've gofmt'ed the code now.

**Comment by brikis98** [comment]
Re-running tests!

**Comment by brikis98** [comment]
Hm, there were a number of test failures that seem to be related to waiting on pods...

Could you look into:

* `TestWaitUntilPodAvailableReturnsSuccessfully`
* `TestWaitUntilPodWithMultipleContainersAvailableReturnsSuccessfully`
* `TestWaitUntilPodAvailableWithReadinessProbe`
* `TestTunnelOpensAPortForwardTunnelToPod`
* `TestTunnelOpensAPortForwardTunnelToService`
* `TestRemoteChartInstall`
* `TestRemoteChartInstallUpgradeRollback`

**Comment by Hsn723** [comment]
@brikis98 Sorry, it looks like I misunderstood the documentation. Under Kubernetes v1.15 which is used in your CI, 
 `pod.Status.ContainerStatus.Started` is always `nil`. The reason for that is likely that startupProbes haven't been introduced yet (the documentation only showed the latest state of the Pod API...).

I've fixed it now (plus a typo with a test case I introduced), please take a look.

**Review by brikis98 — APPROVED** [review]
LGTM! I'll kick off tests again.

**Comment by brikis98** [comment]
Oh, actually, could you pull the latest from `master` first?

**Comment by Hsn723** [comment]
Sure thing. I've synced with `master`

**Comment by brikis98** [comment]
Thanks! I'll kick off tests shortly.

**Review by brikis98 — APPROVED** [review]
Two tests failed, but it seems to be some kind of transient issue with GCP compute instances, so I'm going to merge. Thanks!

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.30.27

**Comment by Hsn723** [comment]
Thank you!

**Pull request merged**